### PR TITLE
Initial support for rendering with Skia and Vulkan

### LINKED ISSUE #1546: Skia: Add support for using Vulkan (optional)
Since Skia can render with Vulkan, it would be nice if we offered this as an option.

### PR DESCRIPTION
**Pull request by tronical**
Closes #1546

### CODE REVIEW

**Comment by tronical** [comment]
The build issue is the old issue of not being able to build Skia from source on Windows in the CI. Happens because be build with all features :(

File: internal/renderers/skia/vulkan_surface.rs [hunk 1]
```diff
@@ -0,0 +1,427 @@
...
+        let image_view = self.swapchain_image_views.borrow()[image_index as usize].clone();
+        let image_object = image_view.image();
+
+        let format = image_view.format();
+
+        let (vk_format, color_type) = match format {
+            Some(vulkano::format::Format::B8G8R8A8_UNORM) => {
+                (skia_safe::gpu::vk::Format::B8G8R8A8_UNORM, skia_safe::ColorType::BGRA8888)
+            }
+            _ => unimplemented!(),
```
Comment (tronical) [inline internal/renderers/skia/vulkan_surface.rs:285]: We can't run into this without the  constructor failing first. But I can change this into a debug_assert.

**Comment by tronical** [comment]
This is back to a draft. The build system issue on Windows needs solving first.

**Comment by tronical** [comment]
Found what looks like a solution for the build issue.

**Review by ogoffart — APPROVED** [review]

File: .github/actions/setup-rust/action.yaml [hunk 2]
```diff
@@ -32,6 +32,7 @@ runs:
       if: runner.os == 'Windows'
       run: |
         echo "RUSTFLAGS=-C debuginfo=0" | Out-File -FilePath $env:GITHUB_ENV -Encoding utf8 -Append
+        git config --system core.longpaths true
```
[the remark this answers is not in the document]
Comment (tronical) [reply .github/actions/setup-rust/action.yaml:39]: Sure, I'll add a comment. It's a feature that's not in git but in the fork of git used for Windows.

**Pull request merged**